docs/content/ChoiceInputField.mdx

```diff
@@ -0,0 +1,155 @@
+---
+title: ChoiceInputField
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/ChoiceInputField.tsx
+storybook: '/react/storybook?path=/story/forms-choiceinputfield--checkbox-input-field'
+---
+
+<!--
+TODO: add `validationStatus` examples when we have the validation styling done
+-->
+
+import {ComponentChecklist} from '../src/component-checklist'
+import {ChoiceInputField, Checkbox, Radio} from '@primer/react'
+import {MarkGithubIcon} from '@primer/octicons-react'
+
+The `ChoiceInputField` component is used to render a labelled checkbox or radio input and, optionally, associated hint text and graphic.
+
+## Anatomy
+
+### ChoiceInputField
+
+The container that handles the layout and passes the relevant IDs and ARIA attributes it's children.
+
+`ChoiceInputField.Label` and `ChoiceInputField.Input` are required children.
+
+### ChoiceInputField.Label
+
+A `ChoiceInputField.Label` must be passed, but it may be visually hidden.
+
+### ChoiceInputField.Caption
+
+If this field needs additional context, a `ChoiceInputField.Caption` may be used to render hint text.
+
+### ChoiceInputField.LeadingVisual
+
+If the selectable option would be easier to understand with a visual, the `ChoiceInputField.LeadingVisual` component may be used.
+
+## Examples
+
+### Checkbox
+
+```jsx live
+<ChoiceInputField>
+  <ChoiceInputField.Label>Option one</ChoiceInputField.Label>
+  <Checkbox />
+</ChoiceInputField>
+```
+
+### Radio
+
+```jsx live
+<ChoiceInputField>
+  <ChoiceInputField.Label>Option one</ChoiceInputField.Label>
```

Comment:
Can we add more radios to this example? Just to emphasise that users shouldn't use a single radio ever, but only as part of a set.